From eb26786561bb4d724f98e0fe8569ce8047882381 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:39:08 -0400
Subject: [x86] cpufeature: Rename X86_FEATURE_CLFLSH to X86_FEATURE_CLFLUSH

Message-id: <1406040016-3289-154-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86232
O-Subject: [RHEL7 PATCH v2 153/221] x86, cpufeature: Rename X86_FEATURE_CLFLSH to X86_FEATURE_CLFLUSH
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "H. Peter Anvin" <hpa@linux.intel.com>

We call this "clflush" in /proc/cpuinfo, and have
cpu_has_clflush()... let's be consistent and just call it that.

Cc: Gleb Natapov <gleb@kernel.org>
Cc: Paolo Bonzini <pbonzini@redhat.com>
Cc: Alan Cox <alan@linux.intel.com>
Link: http://lkml.kernel.org/n/tip-mlytfzjkvuf739okyn40p8a5@git.kernel.org
(cherry picked from commit 840d2830e6e56b8fdacc7ff12915dd91bf91566b)

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/include/asm/cpufeature.h b/arch/x86/include/asm/cpufeature.h
index 8035e0c..556aaff 100644
--- a/arch/x86/include/asm/cpufeature.h
+++ b/arch/x86/include/asm/cpufeature.h
@@ -37,7 +37,7 @@
 #define X86_FEATURE_PAT  (0*32+16) /* Page Attribute Table */
 #define X86_FEATURE_PSE36 (0*32+17) /* 36-bit PSEs */
 #define X86_FEATURE_PN  (0*32+18) /* Processor serial number */
-#define X86_FEATURE_CLFLSH (0*32+19) /* "clflush" CLFLUSH instruction */
+#define X86_FEATURE_CLFLUSH (0*32+19) /* CLFLUSH instruction */
 #define X86_FEATURE_DS  (0*32+21) /* "dts" Debug Store */
 #define X86_FEATURE_ACPI (0*32+22) /* ACPI via MSR */
 #define X86_FEATURE_MMX  (0*32+23) /* Multimedia Extensions */
@@ -313,7 +313,7 @@ extern const char * const x86_power_flags[32];
 #define cpu_has_pmm_enabled boot_cpu_has(X86_FEATURE_PMM_EN)
 #define cpu_has_ds  boot_cpu_has(X86_FEATURE_DS)
 #define cpu_has_pebs  boot_cpu_has(X86_FEATURE_PEBS)
-#define cpu_has_clflush  boot_cpu_has(X86_FEATURE_CLFLSH)
+#define cpu_has_clflush  boot_cpu_has(X86_FEATURE_CLFLUSH)
 #define cpu_has_bts  boot_cpu_has(X86_FEATURE_BTS)
 #define cpu_has_gbpages  boot_cpu_has(X86_FEATURE_GBPAGES)
 #define cpu_has_arch_perfmon boot_cpu_has(X86_FEATURE_ARCH_PERFMON)
diff --git a/arch/x86/kernel/cpu/common.c b/arch/x86/kernel/cpu/common.c
index fbc47f6..74a58e1 100644
--- a/arch/x86/kernel/cpu/common.c
+++ b/arch/x86/kernel/cpu/common.c
@@ -1016,7 +1016,7 @@ __setup("show_msr=", setup_show_msr);
 
 static __init int setup_noclflush(char *arg)
 {
- setup_clear_cpu_cap(X86_FEATURE_CLFLSH);
+ setup_clear_cpu_cap(X86_FEATURE_CLFLUSH);
  return 1;
 }
 __setup("noclflush", setup_noclflush);
diff --git a/arch/x86/kernel/smpboot.c b/arch/x86/kernel/smpboot.c
index 70603b9..31aa398 100644
--- a/arch/x86/kernel/smpboot.c
+++ b/arch/x86/kernel/smpboot.c
@@ -1364,7 +1364,7 @@ static inline void mwait_play_dead(void)
 
  if (!this_cpu_has(X86_FEATURE_MWAIT))
   return;
- if (!this_cpu_has(X86_FEATURE_CLFLSH))
+ if (!this_cpu_has(X86_FEATURE_CLFLUSH))
   return;
  if (__this_cpu_read(cpu_info.cpuid_level) < CPUID_MWAIT_LEAF)
   return;
diff --git a/arch/x86/kvm/cpuid.c b/arch/x86/kvm/cpuid.c
index 2e959cb..bb07da2 100644
--- a/arch/x86/kvm/cpuid.c
+++ b/arch/x86/kvm/cpuid.c
@@ -268,7 +268,7 @@ static inline int __do_cpuid_ent(struct kvm_cpuid_entry2 *entry, u32 function,
   F(TSC) | F(MSR) | F(PAE) | F(MCE) |
   F(CX8) | F(APIC) | 0 /* Reserved */ | F(SEP) |
   F(MTRR) | F(PGE) | F(MCA) | F(CMOV) |
-  F(PAT) | F(PSE36) | 0 /* PSN */ | F(CLFLSH) |
+  F(PAT) | F(PSE36) | 0 /* PSN */ | F(CLFLUSH) |
   0 /* Reserved, DS, ACPI */ | F(MMX) |
   F(FXSR) | F(XMM) | F(XMM2) | F(SELFSNOOP) |
   0 /* HTT, TM, Reserved, PBE */;
diff --git a/drivers/gpu/drm/gma500/mmu.c b/drivers/gpu/drm/gma500/mmu.c
index 49bac41..c3e67ba 100644
--- a/drivers/gpu/drm/gma500/mmu.c
+++ b/drivers/gpu/drm/gma500/mmu.c
@@ -520,7 +520,7 @@ struct psb_mmu_driver *psb_mmu_driver_init(uint8_t __iomem * registers,
 
  driver->has_clflush = 0;
 
- if (boot_cpu_has(X86_FEATURE_CLFLSH)) {
+ if (boot_cpu_has(X86_FEATURE_CLFLUSH)) {
   uint32_t tfms, misc, cap0, cap4, clflush_size;
 
   /*
-- 
1.7.1